# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/firefox/hotfix-x86-shr.patch
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- firefox-117.0/media/ffvpx/libavcodec/x86/mathops.h.vanilla	2023-09-11 23:28:33.371061975 +0200
+++ firefox-117.0/media/ffvpx/libavcodec/x86/mathops.h	2023-09-11 23:30:43.228053437 +0200
@@ -122,11 +122,7 @@
 
 #define NEG_USR32 NEG_USR32
 static inline uint32_t NEG_USR32(uint32_t a, int8_t s){
-    __asm__ ("shrl %1, %0\n\t"
-         : "+r" (a)
-         : "ic" ((uint8_t)(-s))
-    );
-    return a;
+    return a >> (-s);
 }
 
 #endif /* HAVE_INLINE_ASM */
